src/libexpr/include/nix/expr/eval.hh

```diff
@@ -488,8 +489,11 @@ public:
 
     /**
      * Allow access to a path.
+     *
+     * Only for restrict eval: pure eval just whitelist store paths,
+     * never arbitrary paths.
```

Comment:
The implementation doesn't yet enforce this, but I think this does track for the one place this is in fact used now.